src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -218,12 +219,17 @@ class DateColumnType(val time: Boolean): ColumnType() {
         }
 
         return if (time)
-            "'${DEFAULT_DATE_TIME_STRING_FORMATTER.print(dateTime.toDateTime(DateTimeZone.getDefault()))}'"
+            if (withTimezone) {
+                "'${DEFAULT_DATE_TIME_ZONE_STRING_FORMATTER.print(dateTime)}'"
+            } else {
+                "'${DEFAULT_DATE_TIME_STRING_FORMATTER.print(dateTime.toDateTime(DateTimeZone.getDefault()))}'"
+            }
         else
             "'${DEFAULT_DATE_STRING_FORMATTER.print(dateTime)}'"
     }
 
     override fun valueFromDB(value: Any): Any = when(value) {
+        is DateTime -> value
```

Comment:
The only "non-obvious" change this makes is that the returned instance is the same one, but in all other branches we create a new `DateTime` object. Not sure if that behavior needs to be preserved or not.